src/main/java/cms/gov/madie/measure/services/ActionLogService.java

```diff
@@ -85,4 +86,25 @@ public boolean logMeasureSetAction(
             .build(),
         collection);
   }
+
+  public List<Action> findMeasureHistory(String measureId, String measureSetId) {
+    ActionLog measureActionLogs = measureActionLogRepository.findByTargetId(measureId);
+    Optional<MeasureSetActionLog> measureSetActionLogs =
```

Comment:
It's a little odd that only one of the repos returns an `Optional`. Would it be problematic to have `measureActionLogRepository.findByTargetId` also return an `Optional`?

Reply:
I had the same question but this was something which was already there so I didn't make any change at that time, but yeah now I have fixed as you also suggested that